### ISSUE TITLE
Issue with training loss

### ISSUE THREAD

**Issue by Nabeel965** [phanideepgampa/stacked-capsule-networks#2]
Hi,

I am trying to run your code but it seems training loss is not decreasing. Does it only fluctuate around this value or will this decrease to optimal value after some epochs? 

![image](https://user-images.githubusercontent.com/17645444/75549041-8c8a9f80-5a50-11ea-99ce-223a401578af.png)

**Comment by phanideepgampa** [comment]
It will decrease, but it takes a lot of epochs to train. But, as pointed out by a user, there is an issue with gradient flow through the templates and transformation part.

**Comment by phanideepgampa** [comment]
The code is updated, you could try it now.

**Comment by Harry-Up** [comment]
> It will decrease, but it takes a lot of epochs to train. But, as pointed out by a user, there is an issue with gradient flow through the templates and transformation part.

Sorry, I'm late to update the version that preserves the gradient flow. However, the loss still decreases slowly. Frankly speaking, it reduces 1-3 each 100 batches, while the img_lik is more than 3000. It's hopeless to wait for this loss to fall. 

I try to only optimize the _pcae_ and get the almost same result. So I am thinking about the image likelihood. I have tried to weighted sum the transformed templets with d_m, and get the MSE loss. Its optimization seems much better. The loss value quickly decreases from 4 to 0.5. However, it fluctuates around 0.5, I could not tell whether it has been well trained. 

Is there any idea? :)
